stdlib/mexpr/ast.mc

```diff
@@ -259,9 +255,12 @@ end
 -- PATTERNS --
 --------------
 
+type PatName
+con PName     : Name -> PatName
+con PWildcard : ()   -> PatName
```

Comment:
Should this perhaps be MLang syntax instead? (That is part of `VarPat`)

Reply:
Once we get a typesystem this choice will semi-significantly affect how they can be used. The last we got to (as I recall, @EliasC can probably clarify/remind) was essentially that types outside mlang are closed while those inside are extensible. This means that as they are now it would be sufficient to pattern match on `PVar { ident = PName _}` and `PVar { ident = PWildcard _}`, while if they were defined in mlang there would have to be an additional `PVar { ident = _ }` pattern.

(Note that it would be fine if the semantic function consumed a `PatName` directly, it's the nesting that means that we want to ensure exhaustiveness in this way)

Reply:
I put it outside here since I assumed this is a type that would not get extended. Let's discuss at the next meeting!